crates/matrix-sdk-common/src/linked_chunk/relational.rs

```diff
@@ -310,6 +310,17 @@ impl<Item, Gap> RelationalLinkedChunk<Item, Gap> {
             }
         })
     }
+
+    /// Return an iterator over all items.
+    pub fn items(&self) -> impl Iterator<Item = (Position, &Item, &RoomId)> {
+        self.items.iter().filter_map(|item_row| {
+            if let Either::Item(item) = &item_row.item {
+                Some((item_row.position, item, item_row.room_id.as_ref()))
+            } else {
+                None
+            }
+        })
+    }
```

Comment:
Seems this is `unordered_items` with an extra return value; can we fuse those two functions?

Reply:
oh i found the difference now, `unordered_items` is keyed by room. Maybe the names could be clearer…